core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -609,15 +609,15 @@ private void finishDoStop(CompletableFuture<Void> future) {
                 shutdownExecutor(config.blockingTaskExecutor());
             }
 
-            Stream.concat(config.virtualHosts()
-                                .stream()
-                                .filter(VirtualHost::shutdownBlockingTaskExecutorOnStop)
-                                .map(VirtualHost::blockingTaskExecutor),
-                          config.serviceConfigs()
-                                .stream()
-                                .filter(ServiceConfig::shutdownBlockingTaskExecutorOnStop)
-                                .map(ServiceConfig::blockingTaskExecutor))
-                  .forEach(this::shutdownExecutor);
+            shutdownExecutor(Stream.concat(config.virtualHosts()
```

Comment:
You could pass `ServerConfig.blockingTaskExecutor()` together to make the shutdown process even faster.